maap/dps/dps_job.py

```diff
@@ -63,7 +63,7 @@ def __init__(self, not_self_signed=True):
         self.__metrics = dict()
         
     def retrieve_status(self):
-        url = os.path.join(ConfigReader().dps_job, self.id, endpoints.DPS_JOB_STATUS)
+        url = os.path.join(MaapConfig().dps_job, self.id, endpoints.DPS_JOB_STATUS)
```

Comment:
Add `maap: MAAP` to the `__init__` method and set `self.maap = maap`, then you can replace `MaapConfig()` with `self.maap.config` everywhere. Also, when building a URL, use `urllib.parse.urljoin` (or `urlunsplit`), not `os.path.join`:

```suggestion
        url = urllib.parse.urljoin(self.maap.config.dps_job, f"{self.id}/{endpoints.DPS_JOB_STATUS}")
```